### PR TITLE
Etcd prereq fixes

### PR DESCRIPTION
**Pull request by Arau**
I found a few issues with content misplaced, lines not following our best practices directives, some bash lines that could do with wrapping, etc. So I made a few changes to address style. 

On the other hand, I believe that having 2 prereq pages for the same is contradictory. A prerequisites list must be able to be followed as a check list so a user can checkbox the items checking their cluster. Because of that, I suggest the structure with an etcd dir and then, the preferred page and the secondary option. Where it is clear what is what. 

In addition, I put the EBS gp3 specifics into the EKS install page. Otherwise, we should be explaining what SC to use on every cloud provider in the prerequisites page. 
For AWS, we put in place the SC example, I believe that in a different PR, we should do that for all cloud providers.

### CODE REVIEW

File: docs/install/amazon-web-services-eks.md [hunk 1]
```diff
@@ -111,37 +111,34 @@ kubectl storageos preflight
...
-> 💡 The default `StorageClass` in EKS is `gp2` which is not recommended, instead we will create a `gp3` `StorageClass` and set it as default, at least until we install Ondat:
-
-```bash
-kubectl create -f - <<EOF
-kind: StorageClass
-apiVersion: storage.k8s.io/v1
-metadata:
-  name: gp3
-  annotations:
-    storageclass.kubernetes.io/is-default-class: "true"
```
[the remark this answers is not in the document]
Comment (cvlc) [reply docs/install/amazon-web-services-eks.md:127]: I agree, but we still need to create the `StorageClass` if we want to use GP3 for etcd! I'll add a suggestion for that.

**Comment by cannischan** [comment]
Redirects required: 
* https://docs.ondat.io/docs/docs/prerequisites/etcd -> docs/prerequisites/etcd/etcd

* https://docs.ondat.io/docs/prerequisites/etcd-outside-k8s/etcd-outside-the-cluster/ -https://docs.ondat.io/docs/prerequisites/etcd/etcd-outside-the-cluster/

File: docs/prerequisites/etcd/_index.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+title: "Etcd"
+---
+
+Ondat requires an etcd cluster in order to function. For more information on
+why etcd is required, see our [etcd concepts](/docs/concepts/etcd) page.
+
+The Kubernetes etcd cannot be used for Ondat's configuration as per Kubernetes
+requirements.
```
Comment (cvlc) [inline docs/prerequisites/etcd/_index.md:9]:
```suggestion
standards set out by the Kubernetes project.
```

File: docs/prerequisites/etcd/_index.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+For most use-cases it is recommended installing the Ondat etcd operator, which
```
Comment (cvlc) [inline docs/prerequisites/etcd/_index.md:11]:
```suggestion
For most use-cases it is recommended to install the Ondat etcd operator, which
```

File: docs/prerequisites/etcd/_index.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+will manage creation and maintenance of Ondat's required etcd cluster. In some
+circumstances it makes sense to install etcd on separate machines outside of
```
Comment (cvlc) [inline docs/prerequisites/etcd/_index.md:13]:
```suggestion
circumstances, it makes sense to install etcd on separate machines outside of
```

File: docs/prerequisites/etcd/etcd-outside-the-cluster.md [hunk 5]
```diff
@@ -1,19 +1,23 @@
...
 linkTitle: Etcd outside the cluster
-weight: 600
+weight: 700
 ---
 
+This page documents the process for installing etcd outside the Kubernetes
+cluster.
 
+In some circumstances it can make sense to run etcd outside of Kubernetes. One
+of such circumstances is running an on-premises Kubernetes cluster and not
```
Comment (cvlc) [inline docs/prerequisites/etcd/etcd-outside-the-cluster.md:11]:
```suggestion
example is when running an on-premises Kubernetes cluster, if you do not
```

File: docs/prerequisites/etcd/etcd-outside-the-cluster.md [hunk 6]
```diff
@@ -1,19 +1,23 @@
...
+having access to reliable cloud disks (for storing etcd data).
```
Comment (cvlc) [inline docs/prerequisites/etcd/etcd-outside-the-cluster.md:12]:
```suggestion
have access to reliable cloud disks to provide high availability to etcd data.
```

File: docs/prerequisites/etcd/etcd-outside-the-cluster.md [hunk 7]
```diff
@@ -1,19 +1,23 @@
...
-
-For production installations running etcd outside the cluster, Ondat strongly recommends running etcd on a minimum of 3 dedicated virtual machines. This
+For production installations running etcd outside the cluster, Ondat strongly
+recommends running etcd on a minimum of 3 dedicated virtual machines. This
 topology offers strong guarantees of resilience and uptime.
 
-Ondat doesn't require a high performance etcd cluster, as the throughput
-of metadata to the cluster is low. However, we recommend a careful assessment
-of IOPS capacity [best practices](/docs/operations/etcd/) to ensure that etcd
+Ondat doesn't require a high performance etcd cluster, as the throughput of
```
Comment (cvlc) [inline docs/prerequisites/etcd/etcd-outside-the-cluster.md:18]:
```suggestion
Ondat doesn't require a high performance etcd cluster as the throughput of
```

File: docs/prerequisites/etcd/etcd.md [hunk 8]
```diff
@@ -1,39 +1,37 @@
...
 
 ## Configuring Storage for Etcd
 
-We highly recommend using cloud provider network attached disks for storing etcd data, such as EBS volumes, Google Persistent Disks, Azure Disks, etc. This allows the etcd operator to recover from node failures.
+We highly recommend using cloud provider network attached disks for storing
+etcd data, such as EBS volumes, Google Persistent Disks, Azure Disks, etc. This
+allows the etcd operator to recover from node failures.
 
-For testing environments a node-local storage option can be used, such as [Local Path Provisioner](https://github.com/rancher/local-path-provisioner). This will store etcd data on the node hosting an etcd pod
+For testing environments a node-local storage option can be used, such as
```
Comment (cvlc) [inline docs/prerequisites/etcd/etcd.md:28]:
```suggestion
For testing environments or where there are no resilient storage options available,
a node-local storage option can be used, such as
```

File: docs/prerequisites/etcd/etcd.md [hunk 9]
```diff
@@ -1,39 +1,37 @@
...
-
-In AWS, you can use EBS volumes to host the etcd PVCs. The Ondat etcd usage of disk depends on the size of the Kubernetes cluster. However, it is recommended that the disks have at least 800 IOPS at any point in time. The best cost effective storage class that fulfils such requirements is gp3. If gp2 is used, it is paramount to use a volume bigger than 256Gi as it will have enough IOPS even when the burstable credits are exhausted.
-
-To use a gp3 storage class in Kubernetes it is required to install the Amazon CSI Driver. Follow [this guide](https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html) to install. The procedure is comprehended by the following steps:
-
-* Create IAM permissions <https://docs.aws.amazon.com/eks/latest/userguide/csi-iam-role.html>
-* Install the CSI driver
-  * [Using EKS addon](https://docs.aws.amazon.com/eks/latest/userguide/managing-ebs-csi.html)
-  * [Using self-managed add on](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/install.md) (AWS clusters, but not in EKS)
+> ⚠️ The `local-path` StorageClass is only recommended for **non production**
```
Comment (cvlc) [inline docs/prerequisites/etcd/etcd.md:32]:
```suggestion
> ⚠️ The `local-path` StorageClass is only recommended when other, better
```

File: docs/prerequisites/etcd/etcd.md [hunk 10]
```diff
@@ -1,39 +1,37 @@
...
+> clusters, as this stores all the data of the `etcd` peers locally, which
```
Comment (cvlc) [inline docs/prerequisites/etcd/etcd.md:33]:
```suggestion
> storage classes are not available, as this stores 
> all the data of the `etcd` peers locally, which
```

File: docs/prerequisites/etcd/etcd.md [hunk 11]
```diff
@@ -1,39 +1,37 @@
...
+> makes it susceptible to state being lost on node failures.
```
Comment (cvlc) [inline docs/prerequisites/etcd/etcd.md:34]:
```suggestion
> makes it susceptible to state being lost on node failures.
> In the case of `local-path` storage, a minimum of 5 etcd nodes
> is recommended to increase resilience.
```

File: docs/prerequisites/etcd/etcd.md [hunk 12]
```diff
@@ -45,17 +43,22 @@ An etcd cluster can be created in three different ways:
...
+    --include-etcd \
+    --etcd-storage-class <the storage class you want to use for etcd> \
+    --etcd-tls-enabled
 ```
 
 ### **Configurable:** Manually applying an `etcdcluster` custom resource
 
-This installation method allows the most configuration of the etcd cluster, but is the most error-prone.
+This installation method allows the most configuration of the etcd cluster, but
+is the most error-prone.
```
Comment (cvlc) [inline docs/prerequisites/etcd/etcd.md:61]:
```suggestion
is error-prone and therefore not recommended in situations in which the Helm chart
or plugin can be used, instead.
```

**Comment by cvlc** [comment]
Thanks @Arau - great stuff! I've peppered in some suggestions. There are also some markdown linting errors, it would be ideal if we could address those before merge.

File: docs/prerequisites/etcd/etcd.md (hunk 11 above)
Comment (Mojachieee) [reply docs/prerequisites/etcd/etcd.md:34]: FYI it's recommended to run 5 nodes even when using EBS (or other cloud disks)

**Review by Mojachieee — DISMISSED** [review]
Looks good. Calum makes some good suggestions though

File: docs/prerequisites/etcd/_index.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
+---
+title: "Etcd"
+---
+
+Ondat requires an etcd cluster in order to function. For more information on
+why etcd is required, see our [etcd concepts](/docs/concepts/etcd) page.
+
+The Kubernetes etcd cannot be used for Ondat's configuration as per Kubernetes
```
[the remark this answers is not in the document]
Comment (Arau) [reply docs/prerequisites/etcd/_index.md:8]: fantastic thank you 😃

**Review by cannischan — DISMISSED** [review]

**Review by hubvu — APPROVED** [review]

**Review by cvlc — APPROVED** [review]

**Comment by Arau** [comment]
redirects applied.

**Pull request merged**